README.md
# gify

Make GIF with video and images

## Example ( [Live Demo](https://swimmingkiim.github.io/gif_maker/) )

![createGifFromVideo example](https://github.com/swimmingkiim/blog-comments/blob/main/assets/images/gify_create_gif_from_video_example.gif?raw=true)

## Features

- `getFramesBytes` Get Frame images from video as List<Uint8List>
- `createGifFromVideo` Create GIF from video as Uint8List
- `createGifFromImages` Create GIF from images as Uint8List

## Getting started

### Prerequisites

- `ios`
  - Minimum target version : 12.1
- `android`
  - Minimum SDK version : 24
- `Run example`
  - Example uses `image_picker`, so you need to modify your Info.plist ([documentation](https://pub.dev/packages/image_picker#ios))

### Installation

```bash
flutter pub add gify
```

## Usage

```dart
  Future<List<Uint8List>?> testFrames() async {
    final XFile? videoFile =
        await ImagePicker().pickVideo(source: ImageSource.gallery);
    if (videoFile != null) {
      final result = await _gifyPlugin.getFramesBytes(videoFile, fps:1);
      return result;
    }
    return null;
  }

  Future<Uint8List?> testVideoToGif() async {
    final XFile? videoFile =
        await ImagePicker().pickVideo(source: ImageSource.gallery);
    if (videoFile != null) {
      final result = await _gifyPlugin.createGifFromVideo(videoFile, fps:1, width: 320);
      return result;
    }
    return null;
  }

  Future<Uint8List?> testImagesToGif() async {
    final List<XFile> imageFiles = await ImagePicker().pickMultiImage();
    final result = await _gifyPlugin.createGifFromImages(imageFiles,
        fps: 1,
        width: 300,
        height: 480,
        textMessages: [
          const GifyTextMessage(
            text: 'test111',
            fontColor: Color.fromRGBO(247, 24, 7, 1),
            fontSize: 30,
            x: 10.0,
            y: 10.0,
          ),
          const GifyTextMessage(
            text: 'test222',
            fontColor: Color.fromRGBO(7, 67, 247, 1),
            fontSize: 20,
            x: 30.0,
            y: 30.0,
          ),
        ]);
    return result;
  }
```

## Contributing

- Pull Requests are always welcome!
- Please send a Pull Request to https://github.com/swimmingkiim/gify

## License

- [MIT](https://github.com/swimmingkiim/gify/blob/main/LICENSE)


CHANGELOG.md
## 1.0.7

- Fix size option in `createGifFromImages`
## 1.0.6

- Fix text error in native
## 1.0.5

- Hot Fix error when textMessages field is null in web
## 1.0.4

- Add text features with `GifyTextMessage`
## 1.0.3

- Add width option for `createGifFromVideo` and `createGifFromImages`
- Add height option for `createGifFromVideo` and `createGifFromImages`
- Add forceOriginalAspectRatio option for `createGifFromVideo` and `createGifFromImages`

## 1.0.2

- Fix frame sort order
- Apply fps when generate GIF file in native
- Remove some logs and comments in web

## 1.0.1

- Fix android install error

## 1.0.0

initialize gify package

### New Features

- `getFramesBytes` Get Frame images from video as List<Uint8List>
- `createGifFromVideo` Create GIF from video as Uint8List
- `createGifFromImages` Create GIF from images as Uint8List


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter/services.dart';
import 'dart:async';

import 'package:gify/gify.dart';
import 'package:image_picker/image_picker.dart';

void main() {
  WidgetsFlutterBinding.ensureInitialized();
  runApp(const MyApp());
}

class MyApp extends StatefulWidget {
  const MyApp({super.key});

  @override
  State<MyApp> createState() => _MyAppState();
}

class _MyAppState extends State<MyApp> {
  final _gifyPlugin = Gify();
  Uint8List? bytes;

  @override
  void initState() {
    super.initState();
  }

  Future<List<Uint8List>?> testFrames() async {
    final XFile? videoFile =
        await ImagePicker().pickVideo(source: ImageSource.gallery);
    if (videoFile != null) {
      final result = await _gifyPlugin.getFramesBytes(videoFile, fps: 1);
      // Uncomment below code to see result
      setState(() {
        if (result != null) {
          bytes = result[0];
        }
      });
      return result;
    }
    return null;
  }

  Future<Uint8List?> testVideoToGif() async {
    final XFile? videoFile =
        await ImagePicker().pickVideo(source: ImageSource.gallery);
    if (videoFile != null) {
      final result = await _gifyPlugin
          .createGifFromVideo(videoFile, fps: 1, height: 320, textMessages: [
        const GifyTextMessage(
          text: 'test111',
          fontColor: Color.fromRGBO(247, 24, 7, 1),
          fontSize: 30,
          x: 10.0,
          y: 310.0,
        ),
        const GifyTextMessage(
          text: 'test222',
          fontColor: Color.fromRGBO(7, 67, 247, 1),
          fontSize: 20,
          x: 30.0,
          y: 30.0,
        ),
      ]);
      // Uncomment below code to see result
      setState(() {
        if (result != null) {
          bytes = result;
        }
      });
      return result;
    }
    return null;
  }

  Future<Uint8List?> testImagesToGif() async {
    final List<XFile> imageFiles = await ImagePicker().pickMultiImage();
    final result = await _gifyPlugin.createGifFromImages(imageFiles,
        fps: 1,
        width: 300,
        height: 480,
        textMessages: [
          const GifyTextMessage(
            text: 'test111',
            fontColor: Color.fromRGBO(247, 24, 7, 1),
            fontSize: 30,
            x: 10.0,
            y: 10.0,
          ),
          const GifyTextMessage(
            text: 'test222',
            fontColor: Color.fromRGBO(7, 67, 247, 1),
            fontSize: 20,
            x: 30.0,
            y: 30.0,
          ),
        ]);
    // Uncomment below code to see result
    setState(() {
      if (result != null) {
        bytes = result;
      }
    });
    return result;
  }

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Plugin example app'),
        ),
        body: Column(
          children: [
            Center(
              child: InkWell(
                onTap: () => testFrames(),
                child: const Text('test getFramesBytes from video'),
              ),
            ),
            Center(
              child: InkWell(
                onTap: () => testVideoToGif(),
                child: const Text('test createGifFromVideo'),
              ),
            ),
            Center(
              child: InkWell(
                onTap: () => testImagesToGif(),
                child: const Text('test createGifFromImages'),
              ),
            ),
            if (bytes != null) Image.memory(bytes!)
          ],
        ),
      ),
    );
  }
}
